### PR TITLE
Introduce the repository PEX_TOOL.

### PR DESCRIPTION
**Pull request by jsirois**
This allows inspecting a PEX file's distributions as well as exporting
them as wheels.

### CODE REVIEW

File: tests/tools/commands/test_repository.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Iterator
+
+
+FABRIC_VERSION = "2.5.0"
+
+
```
Comment (Eric-Arellano) [inline tests/tools/commands/test_repository.py:23]: Could you please add a test for the `info` subcommand?

**Review by Eric-Arellano — APPROVED** [review]
Thanks - I'm not seeing the big picture of how this will be used, particularly the extract mode? To allow more precise extraction of distributions from a pex, rather than resolving from the entire thing?

**Comment by jsirois** [comment]
> Thanks - I'm not seeing the big picture of how this will be used, particularly the extract mode? To allow more precise extraction of distributions from a pex, rather than resolving from the entire thing?

It could be used for that over in Pants, but its a good deal slower than `--pex-repository` which was built for that case.  Beside serving as due diligence that this approach is slower, it provides a benefit over `--pex-repository` for interoperability. Now a PEX can be used as a `--find-links` repo directly by `pip` for example. This is a real use-case @kwlzn has and I'm sure others may too,

File: tests/tools/commands/test_repository.py (hunk 1 above)
Comment (jsirois) [reply tests/tools/commands/test_repository.py:23]: Yup, will do.

**Comment by jsirois** [comment]
To be clear, you can use Pex tools against old PEX files.

Build an old PEX file:
```
$ ~/.venv/pex1/bin/pex --version
pex 1.6.12
$ ~/.venv/pex1/bin/pex -D src/ -emain:do requests -oexample.pex-1.6.12.pex
```

Operate with modern PEX tools against it:
```
$ python -mpex --version
2.1.32
$ python -mpex.tools example.pex-1.6.12.pex info -i4 | jq .build_properties
{
  "class": "CPython",
  "pex_version": "1.6.12",
  "platform": "linux_x86_64",
  "version": [
    3,
    9,
    2
  ]
}
$ python -mpex.tools example.pex-1.6.12.pex repository info -vi4
{
    "project_name": "certifi",
    "version": "2020.12.5",
    "requires_python": null,
    "requires_dists": [],
    "location": "/home/jsirois/.pex/installed_wheels/94048505ff00e0333a34be16e10b3cc60246b0ee/certifi-2020.12.5-py2.py3-none-any.whl"
}
{
    "project_name": "chardet",
    "version": "4.0.0",
    "requires_python": "!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,>=2.7",
    "requires_dists": [],
    "location": "/home/jsirois/.pex/installed_wheels/a0e59763b786f0b4ec004f6a2d33b70d7f19cc78/chardet-4.0.0-py2.py3-none-any.whl"
}
{
    "project_name": "idna",
    "version": "2.10",
    "requires_python": "!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,>=2.7",
    "requires_dists": [],
    "location": "/home/jsirois/.pex/installed_wheels/e4ab6e1e65a4bcbef7a98bc980659fa4d05f1c12/idna-2.10-py2.py3-none-any.whl"
}
{
    "project_name": "urllib3",
    "version": "1.26.3",
    "requires_python": "!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,<4,>=2.7",
    "requires_dists": [
        "brotlipy>=0.6.0; extra == \"brotli\"",
        "pyOpenSSL>=0.14; extra == \"secure\"",
        "cryptography>=1.3.4; extra == \"secure\"",
        "idna>=2.0.0; extra == \"secure\"",
        "certifi; extra == \"secure\"",
        "ipaddress; python_version == \"2.7\" and extra == \"secure\"",
        "PySocks!=1.5.7,<2.0,>=1.5.6; extra == \"socks\""
    ],
    "location": "/home/jsirois/.pex/installed_wheels/63ff6f3c5654e71c522bf88ed86a71e7dc5708f9/urllib3-1.26.3-py2.py3-none-any.whl"
}
{
    "project_name": "requests",
    "version": "2.25.1",
    "requires_python": "!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,>=2.7",
    "requires_dists": [
        "chardet<5,>=3.0.2",
        "idna<3,>=2.5",
        "urllib3<1.27,>=1.21.1",
        "certifi>=2017.4.17",
        "pyOpenSSL>=0.14; extra == \"security\"",
        "cryptography>=1.3.4; extra == \"security\"",
        "PySocks!=1.5.7,>=1.5.6; extra == \"socks\"",
        "win-inet-pton; (sys_platform == \"win32\" and python_version == \"2.7\") and extra == \"socks\""
    ],
    "location": "/home/jsirois/.pex/installed_wheels/fe2c5347c4c3023d3dadc2890eba53244ed59f17/requests-2.25.1-py2.py3-none-any.whl"
}
$ python -mpex.tools example.pex-1.6.12.pex repository extract --serve
certifi 2020.12.5: Repacking wheel as /tmp/tmpprqkdrnv/certifi-2020.12.5-py2.py3-none-any.whl...OK
chardet 4.0.0: Repacking wheel as /tmp/tmpprqkdrnv/chardet-4.0.0-py2.py3-none-any.whl...OK
idna 2.10: Repacking wheel as /tmp/tmpprqkdrnv/idna-2.10-py2.py3-none-any.whl...OK
urllib3 1.26.3: Repacking wheel as /tmp/tmpprqkdrnv/urllib3-1.26.3-py2.py3-none-any.whl...OK
requests 2.25.1: Repacking wheel as /tmp/tmpprqkdrnv/requests-2.25.1-py2.py3-none-any.whl...OK
Serving find-links repo of example.pex-1.6.12.pex via /tmp/tmpprqkdrnv at http://localhost:43537
^CShut down server for find links repo at /tmp/tmpprqkdrnv.
$ python -mpex.tools example.pex-1.6.12.pex repository extract --serve --sources
running sdist
running egg_info
creating src/example.pex_1.6.12.egg-info
writing src/example.pex_1.6.12.egg-info/PKG-INFO
writing dependency_links to src/example.pex_1.6.12.egg-info/dependency_links.txt
writing entry points to src/example.pex_1.6.12.egg-info/entry_points.txt
writing requirements to src/example.pex_1.6.12.egg-info/requires.txt
writing top-level names to src/example.pex_1.6.12.egg-info/top_level.txt
writing manifest file 'src/example.pex_1.6.12.egg-info/SOURCES.txt'
package init file 'src/data/__init__.py' not found (or not a regular file)
reading manifest file 'src/example.pex_1.6.12.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/example.pex_1.6.12.egg-info/SOURCES.txt'
warning: sdist: standard file not found: should have one of README, README.rst, README.txt, README.md

running check
warning: check: missing required meta-data: url

warning: check: missing meta-data: either (author and author_email) or (maintainer and maintainer_email) should be supplied

creating example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4
creating example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src
creating example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/data
creating example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying files to example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4...
copying MANIFEST.in -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4
copying setup.cfg -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4
copying setup.py -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4
copying src/__main__.pyc -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src
copying src/main.py -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src
copying src/main.pyc -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src
copying src/data/url.txt -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/data
copying src/example.pex_1.6.12.egg-info/PKG-INFO -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying src/example.pex_1.6.12.egg-info/SOURCES.txt -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying src/example.pex_1.6.12.egg-info/dependency_links.txt -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying src/example.pex_1.6.12.egg-info/entry_points.txt -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying src/example.pex_1.6.12.egg-info/requires.txt -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying src/example.pex_1.6.12.egg-info/top_level.txt -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
copying src/example.pex_1.6.12.egg-info/zip-safe -> example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/src/example.pex_1.6.12.egg-info
Writing example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4/setup.cfg
Creating tar archive
removing 'example.pex-1.6.12-0.0.0+9ebd3b2b609f17804c51e7707001bc2f2af453c4' (and everything under it)
certifi 2020.12.5: Repacking wheel as /tmp/tmpsxf_vks_/certifi-2020.12.5-py2.py3-none-any.whl...OK
chardet 4.0.0: Repacking wheel as /tmp/tmpsxf_vks_/chardet-4.0.0-py2.py3-none-any.whl...OK
idna 2.10: Repacking wheel as /tmp/tmpsxf_vks_/idna-2.10-py2.py3-none-any.whl...OK
urllib3 1.26.3: Repacking wheel as /tmp/tmpsxf_vks_/urllib3-1.26.3-py2.py3-none-any.whl...OK
requests 2.25.1: Repacking wheel as /tmp/tmpsxf_vks_/requests-2.25.1-py2.py3-none-any.whl...OK
Serving find-links repo of example.pex-1.6.12.pex via /tmp/tmpsxf_vks_ at http://localhost:36141
^CShut down server for find links repo at /tmp/tmpsxf_vks_.
```

File: tests/tools/commands/test_repository.py (hunk 1 above)
Comment (jsirois) [reply tests/tools/commands/test_repository.py]: Done.

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**